chainstate/tx-verifier/src/transaction_verifier/reward_distribution.rs

```diff
@@ -16,30 +16,91 @@
 use static_assertions::assert_eq_size;
 use std::collections::BTreeMap;
 
-use crate::{error::ConnectTransactionError, TransactionSource};
+use crate::error::ConnectTransactionError;
 
 use common::{
     chain::{Block, DelegationId, PoolId},
     primitives::{per_thousand::PerThousand, Amount, Id},
     Uint256,
 };
-use pos_accounting::{
-    AccountingBlockRewardUndo, PoSAccountingOperations, PoSAccountingUndo, PoSAccountingView,
-};
+use pos_accounting::{PoSAccountingOperations, PoSAccountingUndo, PoSAccountingView, PoolData};
 use utils::ensure;
 
-use super::pos_accounting_delta_adapter::PoSAccountingDeltaAdapter;
+pub trait DelegationSharesView {
```

Comment:
I'm not sure about these new traits. Imho it adds complexity to already non-trivial code. Using existing traits even if they are redundant seems like a better option to me.

Reply:
removed the new traits, now using the existing ones, but PoSAccountingOperations needs a generic to change the Undos return type

Reply:
looking good, thanks